apps/frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -1,19 +1,24 @@
 import { type ReactNode } from "react";
 import { Navigate } from "react-router-dom";
 
-import { AppRoute } from "~/libs/enums/app-route.enum.js";
+import { Loader } from "~/libs/components/components.js";
+import { AppRoute, DataStatus } from "~/libs/enums/enums.js";
 import { useAppSelector } from "~/libs/hooks/hooks.js";
 
 type Properties = {
 	children: ReactNode;
 };
 
 const ProtectedRoute = ({ children }: Properties): JSX.Element => {
-	const authenticatedUser = useAppSelector(
-		(state) => state.auth.authenticatedUser,
-	);
+	const { authenticatedUser, dataStatus } = useAppSelector(({ auth }) => auth);
 
 	const hasAuthenticatedUser = Boolean(authenticatedUser);
+	const isLoading =
+		dataStatus === DataStatus.PENDING || dataStatus === DataStatus.IDLE;
```

Comment:
```suggestion
	const isLoading = [DataStatus.PENDING, DataStatus.IDLE].includes(dataStatus);
```

Reply:
There is an error `Argument of type 'ValueOf<{ readonly FULFILLED: "fulfilled"; readonly IDLE: "idle"; readonly PENDING: "pending"; readonly REJECTED: "rejected"; }>' is not assignable to parameter of type '"idle" | "pending"'.` which I don't know how to solve

Reply:
Okay then leave it as is )